session/session.go

```diff
@@ -540,7 +541,31 @@ func (s *session) CommitTxn(ctx context.Context) error {
 
 	var commitDetail *execdetails.CommitDetails
 	ctx = context.WithValue(ctx, execdetails.CommitDetailCtxKey, &commitDetail)
-	err := s.doCommitWithRetry(ctx)
+	var err error
+	txnScope := s.GetSessionVars().TxnCtx.TxnScope
+	if txnScope != config.DefTxnScope {
```

Comment:
`txnScope` could also be an empty string, which is equivalent to "global".
Alternatively, you can set it to "global" when it's empty in `SetSystemVar`.